temperature/routers.py

```diff
@@ -0,0 +1,25 @@
+from fastapi import APIRouter, Depends
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from dependencies import get_db
+from settings import settings
+from temperature import crud, schemas
+
+prefix = settings.API_PREFIX
+router = APIRouter()
+
+
+@router.post(f"{prefix}/temperatures/update/", response_model=dict)
+async def update_temperatures(db: AsyncSession = Depends(get_db)):
+    return await crud.update_temperatures(db=db)
+
+
+@router.get(
+    f"{prefix}/temperatures/",
+    response_model=list[schemas.Temperature]
+)
+async def read_all_temperatures(
+    city_id: int | None = None,
```

Comment:
The use of 'int | None' for the 'city_id' parameter requires Python 3.10 or later. Ensure that your environment supports this syntax or use 'Optional[int]' from the 'typing' module for compatibility with earlier versions.